src/Domain/DonationNotifier.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace MatchBot\Domain;
+
+use MatchBot\Application\Assertion;
+
+class DonationNotifier
+{
+    public static function emailCommandForCollectedDonation(Donation $donation): SendEmailCommand
+    {
+        if (! $donation->getDonationStatus()->isSuccessful()) {
+            throw new \RuntimeException("{$donation} is not successful - cannot send success email");
+        }
+
+        $paymentMethodType = $donation->getPaymentMethodType();
+        $emailAddress = $donation->getDonorEmailAddress();
+        $collectedAt = $donation->getCollectedAt();
+
+        Assertion::notNull(
+            $paymentMethodType,
+            "payment method should not be null for successful donation: {$donation}"
+        );
+
+        Assertion::notNull(
+            $emailAddress,
+            "email address should not be null for successful donation: {$donation}"
+        );
+
+        Assertion::notNull(
+            $collectedAt,
+            "collectedAt should not be null for successful donation: {$donation}"
+        );
+
+
+        $campaign = $donation->getCampaign();
+        $charity = $campaign->getCharity();
+
+        $fundingWithdrawalsByType = $donation->getWithdrawalTotalByFundType();
+
+        $matchedAmount = bcadd($fundingWithdrawalsByType['amountMatchedByPledges'], $fundingWithdrawalsByType['amountMatchedByChampionFunds'], 2);
```

Comment:
move this domain logic into the donation class, if there isn't an equivalent already there.

Reply:
hmm so donation does already have `getFundingWithdrawalTotal` which I think should return the same thing https://github.com/thebiggive/matchbot/blob/1f7b529bcd14975eb3bf98a739a663fa5c238a1d/src/Domain/Donation.php#L806

but that's looking at the total of all withdrawals, whereas the existing SF logic calculating totalMatchedAmount in pardotUtils is specifically looking at only pledges and champion funds. Afaik those are the only two types of funds so that's a distinction without an immediate difference and using a total of all is if anything better.

@NoelLH does that seem right to you? I'll switch this over to use the total of everything.

Reply:
Done in 59155cc7

Reply:
Yes that seems right @bdsl, even with topup pledges there are no plans to involve a third Salesforce object; so the sum of the two we have now should be the same as the sum of all funding.